src/magic_folder/test/test_snapshot.py

```diff
@@ -95,6 +115,52 @@ def test_serialize_author(self):
         )
 
 
+class TestRemoteAuthor(AsyncTestCase):
+    """
+    Test serialization (to/from JSON) of RemoteAuthor
+    """
+
+    def setUp(self):
+        """
+        We have Alices's signing+verify key
+        """
+        d = super(TestRemoteAuthor, self).setUp()
+        self.alice = create_local_author("alice")
+        return d
+
+    def test_author_serialize(self):
```

Comment:
What is the advantage of using Hypothesis for this kind of test? (Especially given that every single key is just going to be 32 bytes of random binary data in either case?) That is, won't it just be slower?

Reply:
I guess the advantages are the usual.  Maybe Hypothesis finds a case that wasn't considered and doesn't work (or works now and it prevents a regression in the future).  It also makes the test deterministic since the various values Hypothesis provides aren't actually random whereas I assume SigningKey generation tries really hard to be really random.

I don't think it's a huge deal.  I mentioned it mainly as a gentle push towards more Hypothesis use in general.

Reply:
My context here is that I guess I'm trying to understand when you *wouldn't* use Hypothesis. If every single test used Hypothesis, the test-suite would be quite a bit slower (right?).

Reply:
My intuition is that it would be slower, yea.  I'm not sure how by much.  A lot depends on how expensive your strategies are and some more depends on your settings (eg how many passing examples does Hypothesis try to find before declaring the test a success).  I don't think it's *all* downside.  One Hypothesis test might replace 10 "regular" tests in some cases - or provide 10 tests worth of coverage when you might only have had time to write 5 "regular" tests.

I think we should definitely be sensitive to performance of the test suite.  It would be great if 99% of the magic-folder tests could run in under 10 seconds.  We're certainly a long way from that.  My intuition says that all of the child process (esp Tahoe-LAFS child processes) the test suite runs is currently contributing most to slowness.  It would be pretty great to have some actual measurements for this, of course. :/  We currently actually do gather performance information on a per-test-case basis on CI ... we just don't do anything useful with it afterwards.

I wish the software behind https://speed.pypy.org/ had become popular and widespread and extremely featureful and easy to deploy and maintain...

Reply:
To the broader question of when you wouldn't use Hypothesis ... I don't think I've gotten far enough along on my Hypothesis journey to be confident in answering that.  If a unit has inputs, why not generate the inputs with Hypothesis?  The times when I currently don't use Hypothesis (when I could) are mainly times when I can't figure out how to express any general properties for the unit under test.  Otherwise, I try to use Hypothesis.  And I hope that I get lucky and in about another year I understand property testing well enough to have a more refined answer here.